src/answers.sql

```diff
@@ -1,15 +1,250 @@
+FROM
+	(
+		SELECT
+			U.ID
+		FROM
+			USERS U
+			INNER JOIN ACCOUNTS A ON U.ID = A.USER_ID
+		WHERE
+			A.TYPE = 'CURRENT_ACCOUNT'
+		GROUP BY
+			U.ID
+		HAVING
+			COUNT(U.ID) > 1
+	);
 
+--
+--
+--
+--
+--
 -- 3
+--
+SELECT
+	ID,
+	ACCOUNT_ID,
+	TYPE,
+	MOUNT
+FROM
+	ACCOUNTS
+ORDER BY
+	MOUNT DESC
+LIMIT
+	5;
 
+--
+--
+--
+--
 -- 4
+-- I MAKE SURE I USE EVERY ACCOUNT BY UNION ALL!
+-- INNER JOIN MAY LEAVE OUT USERS WITH NO MOVEMENTS
+CREATE
+OR REPLACE FUNCTION GET_ACCOUNTS_FINAL_BALANCE () RETURNS TABLE (ACCOUNT_ID UUID, MONEY NUMERIC(15, 2)) AS $$ BEGIN RETURN QUERY WITH accounts_money_movements AS (
+        (
+            -- account_from transactions
+            SELECT account_from AS account_id,
+                CASE
+                    WHEN type = 'IN' THEN mount
+                    ELSE mount * -1
+                END AS mount
+            FROM movements
+        )
+        UNION ALL
+        (
+            -- account_to transactions
+            SELECT CASE
+                    WHEN type = 'TRANSFER' THEN account_to
+                    ELSE account_from
+                END AS account_id,
+                CASE
+                    WHEN type = 'TRANSFER' THEN mount
+                    ELSE 0
+                END AS mount
+            FROM movements
+        )
+        UNION ALL
+        (
+            -- accounts starting money
+            SELECT id AS account_id,
+                mount
+            FROM accounts
+        )
+    )
+SELECT am.account_id,
+    ROUND(SUM(am.mount)::NUMERIC, 2) AS money
+FROM accounts_money_movements am
+GROUP BY am.account_id;
+END;
+$$ LANGUAGE PLPGSQL;
 
+--
+-- Solution applying function  get_accounts_final_balance
+--
+SELECT
+	U.ID,
+	U.NAME,
+	SUM(AB.MONEY) ALL_MONEY
+FROM
+	GET_ACCOUNTS_FINAL_BALANCE () AB
+	INNER JOIN ACCOUNTS A ON AB.ACCOUNT_ID = A.ID
+	INNER JOIN USERS U ON U.ID = A.USER_ID
+GROUP BY
+	U.ID,
+	U.NAME
+ORDER BY ALL_MONEY DESC
+LIMIT 3;
+
+--
+--
+--
 -- 5
+--
+DO $$
+DECLARE
+    account_balance_a DOUBLE PRECISION;
+    account_balance_b DOUBLE PRECISION;
+	
+    account_id_a UUID := '3b79e403-c788-495a-a8ca-86ad7643afaf';
+    account_id_b UUID := 'fd244313-36e5-4a17-a27c-f8265bc46590';
+BEGIN
+
+	--a. First, get the ammount for the account `3b79e403-c788-495a-a8ca-86ad7643afaf` and `fd244313-36e5-4a17-a27c-f8265bc46590` after all their movements.
+	
+	SELECT money into account_balance_a
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_a;
+
+	SELECT money into account_balance_b
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_b;
+
+	RAISE NOTICE 'Account balance from % is %', account_id_a, account_balance_a;
+	RAISE NOTICE 'Account balance from % is %', account_id_b, account_balance_b;
+
+	--b. Add a new movement with the information: from: 3b79e403-c788-495a-a8ca-86ad7643afaf make a transfer to fd244313-36e5-4a17-a27c-f8265bc46590 mount: 50.75
+	IF account_balance_a < 50.75 THEN
```

Comment:
You could have store 50.75 in a variable, so you don't need to update all places if you need to change the amount